<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QWidget" name="m_gButtons" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Ignored">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="m_btnStatus">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&amp;Status</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_btnSources">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>GPSD
Se&amp;rvers</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_btnTargets">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&amp;Targets</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_btnAbout">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Ab&amp;out</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>539</width>
        <height>460</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QStackedWidget" name="m_gStack">
         <widget class="QWidget" name="m_gStatus">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string>HTTP server</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QWidget" name="widget_2" native="true">
                <layout class="QFormLayout" name="formLayout_4">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_16">
                   <property name="text">
                    <string>Status:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLabel" name="m_gHttpActive"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_12">
                   <property name="text">
                    <string>IP address:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLabel" name="m_gHttpAddress"/>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_14">
                   <property name="text">
                    <string>Port:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLabel" name="m_gHttpPort"/>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_18">
                   <property name="text">
                    <string>Requests:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLabel" name="m_gHttpRequests"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="widget" native="true">
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="m_btnHttpStart">
                   <property name="toolTip">
                    <string>Start HTTP server</string>
                   </property>
                   <property name="text">
                    <string>Start server</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="m_btnHttpStop">
                   <property name="toolTip">
                    <string>Stop HTTP server</string>
                   </property>
                   <property name="text">
                    <string>Stop server</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_2">
             <property name="title">
              <string>GPSD clients</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QScrollArea" name="m_gGpsdClientMonitorsParent">
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="widgetResizable">
                 <bool>true</bool>
                </property>
                <widget class="QWidget" name="m_gGpsdClientMonitors">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>0</y>
                   <width>501</width>
                   <height>300</height>
                  </rect>
                 </property>
                 <layout class="QHBoxLayout" name="m_gGpsdClientMonitorsLayout">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="m_gSources">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QTabWidget" name="m_gSourcesTabs">
             <property name="tabsClosable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="m_gTargets">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string>HTTP server</string>
             </property>
             <layout class="QFormLayout" name="formLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>IP address:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="m_gHttpSetIp">
                <property name="toolTip">
                 <string>Interface IP address or &quot;local&quot; or &quot;local6&quot; or &quot;any&quot; (default)</string>
                </property>
                <property name="placeholderText">
                 <string>Interface IP address or &quot;local&quot; or &quot;local6&quot; or &quot;any&quot; (default)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_15">
                <property name="text">
                 <string>Port:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="m_gHttpSetPort">
                <property name="toolTip">
                 <string>Default value: 42317
= gpsd_javascript_relay % (65535-1024) + 1024
= 4773528272747873529 % 64511 + 1024</string>
                </property>
                <property name="accelerated">
                 <bool>true</bool>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>65535</number>
                </property>
                <property name="value">
                 <number>42317</number>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Auto start:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QCheckBox" name="m_gHttpAutoStart">
                <property name="toolTip">
                 <string>Automatically start HTTP server</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>JavaScript and JSON targets</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QTabWidget" name="m_gTargetsTabs">
                <property name="tabsClosable">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="m_gAbout">
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string notr="true">&lt;h1&gt;GPSD to JavaScript relay&lt;/h1&gt;</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Author:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string notr="true">&lt;a href=&quot;https://github.com/EstevanTH/&quot;&gt;Mohamed RACHID&lt;/a&gt;</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>IDE:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string notr="true">&lt;a href=&quot;https://www.qt.io/ide/&quot;&gt;Qt Creator&lt;/a&gt;</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Tested servers:</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string notr="true">&lt;a href=&quot;https://play.google.com/store/apps/details?id=com.jillybunch.shareGPS&quot;&gt;Share GPS (Android)&lt;/a&gt;</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>License:</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string notr="true">LGPL</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Documentation:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string notr="true">&lt;a href=&quot;http://www.catb.org/gpsd/gpsd_json.html&quot;&gt;gpsd_json - gpsd request/response protocol&lt;/a&gt;&lt;br/&gt;
&lt;a href=&quot;http://www.jillybunch.com/sharegps/mobile/gpsd-commands.html&quot;&gt;GPSD server command interface&lt;/a&gt;</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>A small tool producing JS &amp; JSON output based on GPS data provided by a GPSD server</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QLabel" name="m_gBuild">
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_21">
             <property name="text">
              <string>Version:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
